elastalert/elastalert.py

```diff
@@ -440,11 +443,16 @@ def run_rule(self, rule, endtime, starttime=None):
             logging.warning("Attempted to use query start time in the future (%s), sleeping instead" % (starttime))
             return 0
 
-        # Run the rule
-        # If querying over a large time period, split it up into chunks
+        # Run the rule. If querying over a large time period, split it up into segments
+        # The segment size is either buffer_size for normal queries or run_every for
+        # count style queries
         self.num_hits = 0
-        while endtime - rule['starttime'] > self.run_every:
-            tmp_endtime = rule['starttime'] + self.run_every
+        if not rule.get('use_count_query') and not rule.get('use_terms_query'):
```

Comment:
How about extracting this if/else to a separate method? It could have 2 returns (thus no need for `else`) and `segment_size` could take a value of what is returned from that method.